source/joker_effects.c

```diff
@@ -458,27 +505,50 @@ static JokerEffect acrobat_joker_effect(Joker *joker, Card *scored_card) {
     return effect;
 }
 
-static JokerEffect the_duo_joker_effect(Joker *joker, Card *scored_card) {
+__attribute__((unused))
+static JokerEffect hanging_chad_joker_effect(Joker *joker, Card *scored_card, int scored_when) {
     JokerEffect effect = {0};
-    if (scored_card != NULL)
-        return effect; // if card != null, we are not at the end-phase of scoring yet
+
+    switch (scored_when) {
+        case JOKER_CALLBACK_ON_HAND_SCORED_END:
+            joker->data = 2;
+            break;
+        // No need to check if this is the first card scored or not
+        // joker->data will always reach 0 on the first card, then retrigger
+        // will be false and scoring will go onto the next card
+        case JOKER_CALLBACK_ON_CARD_SCORED:
+            effect.retrigger = (joker->data > 0);
+            joker->data -= 1;
+            break;
+    }
+
+    return effect;
+}
+
+static JokerEffect the_duo_joker_effect(Joker *joker, Card *scored_card, int scored_when) {
+    JokerEffect effect = {0};
+
+    SCORE_INDEPENDANT_ONLY(scored_card, scored_when, effect)
     
-     // This is really inefficient but the only way at the moment to check for whole-hand conditions
+    // This is really inefficient but the only way at the moment to check for whole-hand conditions
     u8 suits[NUM_SUITS];
     u8 ranks[NUM_RANKS];
     get_played_distribution(ranks, suits);
 
-    if (hand_contains_n_of_a_kind(ranks) >= 2)
+    if (hand_contains_n_of_a_kind(ranks) >= 2) {
         effect.xmult = 2;
+    }
+
     return effect;
  }
 
-static JokerEffect the_trio_joker_effect(Joker *joker, Card *scored_card) {
+
+static JokerEffect the_trio_joker_effect(Joker *joker, Card *scored_card, int scored_when) {
     JokerEffect effect = {0};
-    if (scored_card != NULL)
-        return effect; // if card != null, we are not at the end-phase of scoring yet
-    
-     // This is really inefficient but the only way at the moment to check for whole-hand conditions
+
+    SCORE_INDEPENDANT_ONLY(scored_card, scored_when, effect)
```

Comment:
```suggestion
    SCORE_ON_EVENT_ONLY(JOKER_CALLBACK_INDEPENDANT, scored_when, effect)
```

Batch - Consolidate joker event restriction macros